cmd/podmanV2/images/search.go

```diff
@@ -0,0 +1,157 @@
+type searchOptionsWrapper struct {
+	entities.ImageSearchOptions
+	// CLI only flags
+	TLSVerifyCLI bool   // Used to convert to an optional bool later
+	Format       string // For go templating
+}
+
+var (
+	searchOptions     = searchOptionsWrapper{}
+	searchDescription = `Search registries for a given image. Can search all the default registries or a specific registry.
+
+	Users can limit the number of results, and filter the output based on certain conditions.`
+
+	// Command: podman search
+	searchCmd = &cobra.Command{
+		Use:     "search [flags] TERM",
+		Short:   "Search registry for image",
+		Long:    searchDescription,
+		PreRunE: preRunE,
+		RunE:    imageSearch,
+		Args:    cobra.ExactArgs(1),
+		Example: `podman search --filter=is-official --limit 3 alpine
+  podman search registry.fedoraproject.org/  # only works with v2 registries
+  podman search --format "table {{.Index}} {{.Name}}" registry.fedoraproject.org/fedora`,
+	}
+
+	// Command: podman image search
+	imageSearchCmd = &cobra.Command{
+		Use:     searchCmd.Use,
+		Short:   searchCmd.Short,
+		Long:    searchCmd.Long,
+		PreRunE: searchCmd.PreRunE,
+		RunE:    searchCmd.RunE,
+		Args:    searchCmd.Args,
+		Example: `podman image search --filter=is-official --limit 3 alpine
+		podman image search registry.fedoraproject.org/  # only works with v2 registries
+		podman image search --format "table {{.Index}} {{.Name}}" registry.fedoraproject.org/fedora`,
+	}
+)
+
+func init() {
+	// search
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: searchCmd,
+	})
+
+	searchCmd.SetHelpTemplate(registry.HelpTemplate())
+	searchCmd.SetUsageTemplate(registry.UsageTemplate())
+	flags := searchCmd.Flags()
+	searchFlags(flags)
+
+	// images search
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: imageSearchCmd,
+		Parent:  imageCmd,
+	})
+
+	imageSearchFlags := imageSearchCmd.Flags()
+	searchFlags(imageSearchFlags)
+}
+
+// searchFlags set the flags for the pull command.
+func searchFlags(flags *pflag.FlagSet) {
+	flags.StringSliceVarP(&searchOptions.Filters, "filter", "f", []string{}, "Filter output based on conditions provided (default [])")
+	flags.StringVar(&searchOptions.Format, "format", "", "Change the output format to a Go template")
+	flags.IntVar(&searchOptions.Limit, "limit", 0, "Limit the number of results")
+	flags.BoolVar(&searchOptions.NoTrunc, "no-trunc", false, "Do not truncate the output")
+	flags.StringVar(&searchOptions.Authfile, "authfile", buildahcli.GetDefaultAuthFile(), "Path of the authentication file. Use REGISTRY_AUTH_FILE environment variable to override")
+	flags.BoolVar(&searchOptions.TLSVerifyCLI, "tls-verify", true, "Require HTTPS and verify certificates when contacting registries")
+	if registry.IsRemote() {
+		_ = flags.MarkHidden("authfile")
```

Comment:
Is there a reason this is hidden?  Do we need a way to handle this in remote?  With the new API is there a way to pass this up?

Reply:
It's hidden in other commands and in v1 as well.

There is currently now way to pass this in v2. The Docker API does that via base64 encoded authfile in the http header but we don't support that yet. I think we need to support it not only for the Docker endpoints but also for us. I suggest to cover that with remote login/logout as it really needs to be consistent across all endpoints that deal with authenticate.

Can we get this PR in?